.github/workflows/ci-go.yml

```diff
@@ -23,12 +30,20 @@ jobs:
         with:
           go-version-file: 'go.work'
           check-latest: true
+          cache: false
 
       - name: 'Test'
-        run: make test
+        uses: robherley/go-test-action@v0.2.0
```

Comment:
what was wrong with the current strategy using the make target?

Reply:
Nothing, this one does exactly the same plus the auto-generated summary which can be seen here: https://github.com/Azure/ARO-HCP/actions/runs/8418935166/attempts/1#summary-23050469359 
This add a bit of convenience when looking at test results and figuring out what is wrong. It is my two cents to add convenience compared to grepping the logs for error.

Reply:
I removed the changes to make review easier.